From 67ee3c60f370461208e1b53dc1bacf892c85ab6a Mon Sep 17 00:00:00 2001
From: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Date: Mon, 17 Jan 2022 18:36:39 +0200
Subject: [PATCH 24/25] s32: pinctrl: move MSCR definitions to a single place

There were multiple files that defined macros for the MSCR registers.
This commit moves them to a single location, as part of the pinctrl
dt-bindings.

Issue: ALB-8283

Upstream-Status: Pending 

Signed-off-by: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/include/asm/arch-s32/siul-s32-gen1.h | 429 ++++++++----------
 drivers/gpio/gpio-s32.c                       |   8 +-
 .../dt-bindings/pinctrl/s32-gen1-pinctrl.h    |  19 +-
 3 files changed, 205 insertions(+), 251 deletions(-)

diff --git a/arch/arm/include/asm/arch-s32/siul-s32-gen1.h b/arch/arm/include/asm/arch-s32/siul-s32-gen1.h
index 99fb403739..67f393b91a 100644
--- a/arch/arm/include/asm/arch-s32/siul-s32-gen1.h
+++ b/arch/arm/include/asm/arch-s32/siul-s32-gen1.h
@@ -6,6 +6,8 @@
 #ifndef __ARCH_ARM_MACH_S32G1_SIUL_H__
 #define __ARCH_ARM_MACH_S32G1_SIUL_H__
 
+#include "dt-bindings/pinctrl/s32-gen1-pinctrl.h"
+
 /* SIUL2_0 and SIUL2_1 MSCR specifications as stated in Reference Manual: */
 
 #define SIUL2_0_MSCR_BASE		(SIUL2_0_BASE_ADDR + 0x00000240)
@@ -32,49 +34,6 @@
 #define SIUL2_0_S32R45_MAX_GPIO		102
 #define SIUL2_0_S32G_MAX_GPIO		112
 
-#define SIUL2_MSCR_MUX_MODE(v)		((v) & 0x0000000f)
-#define SIUL2_MSCR_MUX_MODE_ALT0	(0x0)
-#define SIUL2_MSCR_MUX_MODE_ALT1	(0x1)
-#define SIUL2_MSCR_MUX_MODE_ALT2	(0x2)
-#define SIUL2_MSCR_MUX_MODE_ALT3	(0x3)
-#define SIUL2_MSCR_MUX_MODE_ALT4	(0x4)
-#define SIUL2_MSCR_MUX_MODE_ALT5	(0x5)
-#define SIUL2_MSCR_MUX_MODE_ALT6	(0x6)
-
-/* S32-GEN1 SIUL2_MSCR masks */
-#define SIUL2_MSCR_S32_G1_OBE_EN		(1 << 21)
-#define SIUL2_MSCR_S32_G1_OBE(v)		((v) & SIUL2_MSCR_S32_G1_OBE_EN)
-
-#define SIUL2_MSCR_S32_G1_ODE_EN		(1 << 20)
-#define SIUL2_MSCR_S32_G1_ODE(v)		((v) & SIUL2_MSCR_S32_G1_ODE_EN)
-
-#define SIUL2_MSCR_S32_G1_IBE_EN		(1 << 19)
-#define SIUL2_MSCR_S32_G1_IBE(v)		((v) & SIUL2_MSCR_S32_G1_IBE_EN)
-
-#define SIUL2_MSCR_S32_G1_INV_EN		(1 << 17)
-#define SIUL2_MSCR_S32_G1_INV(v)		((v) & SIUL2_MSCR_S32_G1_INV_EN)
-
-
-#define SIUL2_MSCR_S32_G1_SRC(v)		((v) & (7 << 14))
-#define SIUL2_MSCR_S32_G1_SRC_208MHz		(0 << 14)
-#define SIUL2_MSCR_S32_G1_SRC_150MHz		(4 << 14)
-#define SIUL2_MSCR_S32_G1_SRC_100MHz		(5 << 14)
-#define SIUL2_MSCR_S32_G1_SRC_50MHz		(6 << 14)
-#define SIUL2_MSCR_S32_G1_SRC_25MHz		(7 << 14)
-
-#define SIUL2_MSCR_S32_G1_PUE_EN		(1 << 13)
-#define SIUL2_MSCR_S32_G1_PUE(v)		((v) & SIUL2_MSCR_S32_G1_PUE_EN)
-
-#define SIUL2_MSCR_S32_G1_PUS_EN		(1 << 12)
-#define SIUL2_MSCR_S32_G1_PUS(v)		((v) & SIUL2_MSCR_S32_G1_PUS_EN)
-
-#define SIUL2_MSCR_S32_G1_RCVR(v)		((v) & (3 << 10))
-#define SIUL2_MSCR_S32_G1_RCVR_DBL		(0 << 10)
-#define SIUL2_MSCR_S32_G1_RCVR_SNGL		(1 << 10)
-
-#define SIUL2_MSCR_S32_G1_SMC_DIS		(1 << 5)
-#define SIUL2_MSCR_S32_G1_SMC(v)		((v) & SIUL2_MSCR_S32_G1_SMC_DIS)
-
 /* S32-GEN1 UART0 settings */
 /* TXD */
 #define SIUL2_PC09_MSCR_S32_G1_UART0	41
@@ -111,81 +70,81 @@
 
 /* UART MSCR settings */
 #define SIUL2_MSCR_S32_G1_PORT_CTRL_UART_TXD	 \
-	(SIUL2_MSCR_S32_G1_SRC_100MHz |		 \
-	 SIUL2_MSCR_S32_G1_OBE_EN |		 \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_133_1V8_100_3V3_MHZ |		 \
+	 SIUL2_MSCR_S32_G1_OBE |		 \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_PORT_CTRL_UART_RXD	 \
-	(SIUL2_MSCR_S32_G1_SRC_100MHz |		 \
-	 SIUL2_MSCR_S32_G1_IBE_EN |		 \
-	 SIUL2_MSCR_MUX_MODE_ALT2)
+	(SIUL2_MSCR_S32_G1_SRC_133_1V8_100_3V3_MHZ |		 \
+	 SIUL2_MSCR_S32_G1_IBE |		 \
+	 SIUL2_MSCR_S32_G1_MUX_ID_2)
 
 /* UART IMCR mux modes */
 #define SIUL2_IMCR_S32_G1_UART_RXD_to_pad	0
 
 /* S32GEN1 */
 #define SIUL2_MSCR_S32G_G1_PORT_CTRL_UART0_TXD	 \
-	(SIUL2_MSCR_S32_G1_SRC_100MHz |		 \
-	 SIUL2_MSCR_S32_G1_OBE_EN |		 \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_133_1V8_100_3V3_MHZ |		 \
+	 SIUL2_MSCR_S32_G1_OBE |		 \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #if defined(CONFIG_TARGET_TYPE_S32GEN1_SIMULATOR) || \
 	defined(CONFIG_TARGET_TYPE_S32GEN1_EMULATOR)
 #define SIUL2_MSCR_S32G_G1_PORT_CTRL_UART1_TXD	 \
-	(SIUL2_MSCR_S32_G1_SRC_100MHz |		 \
-	 SIUL2_MSCR_S32_G1_OBE_EN |		 \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_133_1V8_100_3V3_MHZ |		 \
+	 SIUL2_MSCR_S32_G1_OBE |		 \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 #else
 #define SIUL2_MSCR_S32G_G1_PORT_CTRL_UART1_TXD	 \
-	(SIUL2_MSCR_S32_G1_SRC_100MHz |		 \
-	 SIUL2_MSCR_S32_G1_OBE_EN |		 \
-	 SIUL2_MSCR_MUX_MODE_ALT2)
+	(SIUL2_MSCR_S32_G1_SRC_133_1V8_100_3V3_MHZ |		 \
+	 SIUL2_MSCR_S32_G1_OBE |		 \
+	 SIUL2_MSCR_S32_G1_MUX_ID_2)
 #endif
 
 #define SIUL2_MSCR_S32G_G1_PORT_CTRL_UART_RXD	 \
-	(SIUL2_MSCR_S32_G1_SRC_100MHz |		 \
-	 SIUL2_MSCR_S32_G1_IBE_EN |		 \
-	 SIUL2_MSCR_MUX_MODE_ALT0)
+	(SIUL2_MSCR_S32_G1_SRC_133_1V8_100_3V3_MHZ |		 \
+	 SIUL2_MSCR_S32_G1_IBE |		 \
+	 SIUL2_MSCR_S32_G1_MUX_ID_0)
 
 /* UART IMCR mux modes */
-#define SIUL2_IMCR_S32G_G1_UART0_RXD_to_pad	(SIUL2_MSCR_MUX_MODE_ALT2)
+#define SIUL2_IMCR_S32G_G1_UART0_RXD_to_pad	(SIUL2_MSCR_S32_G1_MUX_ID_2)
 #if defined(CONFIG_TARGET_TYPE_S32GEN1_SIMULATOR) || \
 	defined(CONFIG_TARGET_TYPE_S32GEN1_EMULATOR)
-#define SIUL2_IMCR_S32G_G1_UART1_RXD_to_pad	(SIUL2_MSCR_MUX_MODE_ALT3)
+#define SIUL2_IMCR_S32G_G1_UART1_RXD_to_pad	(SIUL2_MSCR_S32_G1_MUX_ID_3)
 #elif defined(CONFIG_TARGET_S32G274ARDB)
-#define SIUL2_IMCR_S32G_G1_UART1_RXD_to_pad	(SIUL2_MSCR_MUX_MODE_ALT2)
+#define SIUL2_IMCR_S32G_G1_UART1_RXD_to_pad	(SIUL2_MSCR_S32_G1_MUX_ID_2)
 #else
-#define SIUL2_IMCR_S32G_G1_UART1_RXD_to_pad	(SIUL2_MSCR_MUX_MODE_ALT4)
+#define SIUL2_IMCR_S32G_G1_UART1_RXD_to_pad	(SIUL2_MSCR_S32_G1_MUX_ID_4)
 #endif
 
 /* S32-GEN1 uSDHC settings */
 #define SIUL2_USDHC_S32_G1_PAD_CTRL_BASE \
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_S32_G1_IBE_EN | \
-	 SIUL2_MSCR_S32_G1_PUE_EN | \
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	 SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_IBE | \
+	 SIUL2_MSCR_S32_G1_PUE | \
 	 SIUL2_MSCR_S32_G1_SMC_DIS)
 
 #define SIUL2_USDHC_S32_G1_PAD_CTRL_CLK \
 	(SIUL2_USDHC_S32_G1_PAD_CTRL_BASE | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_USDHC_S32_G1_PAD_CTRL_CMD \
 	(SIUL2_USDHC_S32_G1_PAD_CTRL_BASE | \
-	 SIUL2_MSCR_S32_G1_PUS_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	 SIUL2_MSCR_S32_G1_PUS | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_USDHC_S32_G1_PAD_CTRL_DATA \
 	(SIUL2_USDHC_S32_G1_PAD_CTRL_BASE | \
-	 SIUL2_MSCR_S32_G1_PUS_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	 SIUL2_MSCR_S32_G1_PUS | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_USDHC_S32_G1_PAD_RST \
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_S32_G1_PUE_EN | \
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	 SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_PUE | \
 	 SIUL2_MSCR_S32_G1_SMC_DIS | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 /* I2C settings */
 /* S32GEN1 */
@@ -243,89 +202,89 @@
 #define SIUL2_PC_00_IMCR_S32G_I2C1_SCLK	(566 - 512)
 
 #define SIUL2_MSCR_S32G_I2C_SDA \
-	 (SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_S32_G1_IBE_EN | \
-	 SIUL2_MSCR_S32_G1_ODE_EN)
+	 (SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_IBE | \
+	 SIUL2_MSCR_S32_G1_ODE)
 
 #define SIUL2_MSCR_S32G_I2C_SCLK \
-	 (SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_S32_G1_IBE_EN | \
-	 SIUL2_MSCR_S32_G1_ODE_EN)
+	 (SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_IBE | \
+	 SIUL2_MSCR_S32_G1_ODE)
 
 /* I2C0 - Serial Data Input */
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C0_SDA \
-	(SIUL2_MSCR_MUX_MODE_ALT1 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	 SIUL2_MSCR_S32G_I2C_SDA)
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C0_SDA_ALT2 \
-	(SIUL2_MSCR_MUX_MODE_ALT2 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_2 | \
 	 SIUL2_MSCR_S32G_I2C_SDA)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C0_SDA (SIUL2_MSCR_MUX_MODE_ALT2)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C0_SDA_ALT3 (SIUL2_MSCR_MUX_MODE_ALT3)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C0_SDA (SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C0_SDA_ALT3 (SIUL2_MSCR_S32_G1_MUX_ID_3)
 
 /* I2C0 - Serial Clock Input */
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C0_SCLK \
-	(SIUL2_MSCR_MUX_MODE_ALT1 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	 SIUL2_MSCR_S32G_I2C_SCLK)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C0_SCLK (SIUL2_MSCR_MUX_MODE_ALT2)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C0_SCLK_ALT3 (SIUL2_MSCR_MUX_MODE_ALT3)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C0_SCLK (SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C0_SCLK_ALT3 (SIUL2_MSCR_S32_G1_MUX_ID_3)
 
 /* I2C1 - Serial Data Input */
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C1_SDA \
-	(SIUL2_MSCR_MUX_MODE_ALT1 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	 SIUL2_MSCR_S32G_I2C_SDA)
 
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C1_SDA_ALT3 \
-	(SIUL2_MSCR_MUX_MODE_ALT3 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_3 | \
 	 SIUL2_MSCR_S32G_I2C_SDA)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C1_SDA (SIUL2_MSCR_MUX_MODE_ALT2)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C1_SDA_ALT4 (SIUL2_MSCR_MUX_MODE_ALT4)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C1_SDA (SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C1_SDA_ALT4 (SIUL2_MSCR_S32_G1_MUX_ID_4)
 
 /* I2C1 - Serial Clock Input */
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C1_SCLK \
-	(SIUL2_MSCR_MUX_MODE_ALT1 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	 SIUL2_MSCR_S32G_I2C_SCLK)
 
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C1_SCLK_ALT3 \
-	(SIUL2_MSCR_MUX_MODE_ALT3 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_3 | \
 	 SIUL2_MSCR_S32G_I2C_SCLK)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C1_SCLK (SIUL2_MSCR_MUX_MODE_ALT2)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C1_SCLK_ALT5 (SIUL2_MSCR_MUX_MODE_ALT5)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C1_SCLK (SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C1_SCLK_ALT5 (SIUL2_MSCR_S32_G1_MUX_ID_5)
 
 /* I2C2 - Serial Data Input */
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C2_SDA \
-	(SIUL2_MSCR_MUX_MODE_ALT1 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	 SIUL2_MSCR_S32G_I2C_SDA)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C2_SDA (SIUL2_MSCR_MUX_MODE_ALT2)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C2_SDA (SIUL2_MSCR_S32_G1_MUX_ID_2)
 
 /* I2C2 - Serial Clock Input */
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C2_SCLK \
-	(SIUL2_MSCR_MUX_MODE_ALT1 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	 SIUL2_MSCR_S32G_I2C_SCLK)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C2_SCLK (SIUL2_MSCR_MUX_MODE_ALT2)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C2_SCLK (SIUL2_MSCR_S32_G1_MUX_ID_2)
 
 /* I2C3 - Serial Data Input */
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C3_SDA \
-	(SIUL2_MSCR_MUX_MODE_ALT2 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_2 | \
 	 SIUL2_MSCR_S32G_I2C_SDA)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C3_SDA (SIUL2_MSCR_MUX_MODE_ALT4)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C3_SDA (SIUL2_MSCR_S32_G1_MUX_ID_4)
 
 /* I2C3 - Serial Clock Input */
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C3_SCLK \
-	(SIUL2_MSCR_MUX_MODE_ALT1 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	 SIUL2_MSCR_S32G_I2C_SCLK)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C3_SCLK (SIUL2_MSCR_MUX_MODE_ALT2)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C3_SCLK (SIUL2_MSCR_S32_G1_MUX_ID_2)
 
 /* I2C4 - Serial Data Input */
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C4_SDA \
-	(SIUL2_MSCR_MUX_MODE_ALT1 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	 SIUL2_MSCR_S32G_I2C_SDA)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C4_SDA (SIUL2_MSCR_MUX_MODE_ALT3)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C4_SDA (SIUL2_MSCR_S32_G1_MUX_ID_3)
 
 /* I2C4 - Serial Clock Input */
 #define SIUL2_MSCR_S32G_PAD_CTRL_I2C4_SCLK \
-	(SIUL2_MSCR_MUX_MODE_ALT2 | \
+	(SIUL2_MSCR_S32_G1_MUX_ID_2 | \
 	 SIUL2_MSCR_S32G_I2C_SCLK)
-#define SIUL2_IMCR_S32G_PAD_CTRL_I2C4_SCLK (SIUL2_MSCR_MUX_MODE_ALT3)
+#define SIUL2_IMCR_S32G_PAD_CTRL_I2C4_SCLK (SIUL2_MSCR_S32_G1_MUX_ID_3)
 
 /* S32G-GEN1 ENET settings */
 /* GMAC0 MDC*/
@@ -383,87 +342,87 @@
 #define SIUL2_MSCR_S32_G1_PFE_MAC1_RXD3_IN 864
 
 #define SIUL2_MSCR_S32_G1_ENET_MDC	\
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	 SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_ENET_MDIO	\
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_S32_G1_IBE_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	  SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_IBE | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
-#define SIUL2_MSCR_S32_G1_ENET_MDIO_IN		SIUL2_MSCR_MUX_MODE_ALT2
+#define SIUL2_MSCR_S32_G1_ENET_MDIO_IN		SIUL2_MSCR_S32_G1_MUX_ID_2
 
 #define SIUL2_MSCR_S32_G1_ENET_TX_CLK \
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	  SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
-#define SIUL2_MSCR_S32_G1_ENET_TX_CLK_IN	SIUL2_MSCR_MUX_MODE_ALT2
+#define SIUL2_MSCR_S32_G1_ENET_TX_CLK_IN	SIUL2_MSCR_S32_G1_MUX_ID_2
 
 #define SIUL2_MSCR_S32_G1_ENET_RX_CLK	\
-	 (SIUL2_MSCR_S32_G1_IBE_EN)
+	 (SIUL2_MSCR_S32_G1_IBE)
 
-#define SIUL2_MSCR_S32_G1_ENET_RX_CLK_IN	SIUL2_MSCR_MUX_MODE_ALT2
+#define SIUL2_MSCR_S32_G1_ENET_RX_CLK_IN	SIUL2_MSCR_S32_G1_MUX_ID_2
 
 #define SIUL2_MSCR_S32_G1_ENET_RX_D0	\
-	 (SIUL2_MSCR_S32_G1_IBE_EN)
+	 (SIUL2_MSCR_S32_G1_IBE)
 
-#define SIUL2_MSCR_S32_G1_ENET_RX_D0_IN		SIUL2_MSCR_MUX_MODE_ALT2
+#define SIUL2_MSCR_S32_G1_ENET_RX_D0_IN		SIUL2_MSCR_S32_G1_MUX_ID_2
 
 #define SIUL2_MSCR_S32_G1_ENET_RX_D1	\
-	 (SIUL2_MSCR_S32_G1_IBE_EN)
+	 (SIUL2_MSCR_S32_G1_IBE)
 
-#define SIUL2_MSCR_S32_G1_ENET_RX_D1_IN		SIUL2_MSCR_MUX_MODE_ALT2
+#define SIUL2_MSCR_S32_G1_ENET_RX_D1_IN		SIUL2_MSCR_S32_G1_MUX_ID_2
 
 #define SIUL2_MSCR_S32_G1_ENET_RX_D2	\
-	 (SIUL2_MSCR_S32_G1_IBE_EN)
+	 (SIUL2_MSCR_S32_G1_IBE)
 
-#define SIUL2_MSCR_S32_G1_ENET_RX_D2_IN		SIUL2_MSCR_MUX_MODE_ALT2
+#define SIUL2_MSCR_S32_G1_ENET_RX_D2_IN		SIUL2_MSCR_S32_G1_MUX_ID_2
 
 #define SIUL2_MSCR_S32_G1_ENET_RX_D3	\
-	 (SIUL2_MSCR_S32_G1_IBE_EN)
+	 (SIUL2_MSCR_S32_G1_IBE)
 
-#define SIUL2_MSCR_S32_G1_ENET_RX_D3_IN		SIUL2_MSCR_MUX_MODE_ALT2
+#define SIUL2_MSCR_S32_G1_ENET_RX_D3_IN		SIUL2_MSCR_S32_G1_MUX_ID_2
 
 #define SIUL2_MSCR_S32_G1_ENET_RX_DV	\
-	 (SIUL2_MSCR_S32_G1_IBE_EN)
+	 (SIUL2_MSCR_S32_G1_IBE)
 
-#define SIUL2_MSCR_S32_G1_ENET_RX_DV_IN		SIUL2_MSCR_MUX_MODE_ALT2
+#define SIUL2_MSCR_S32_G1_ENET_RX_DV_IN		SIUL2_MSCR_S32_G1_MUX_ID_2
 
 #define SIUL2_MSCR_S32_G1_ENET_RX_ER	\
-	 (SIUL2_MSCR_S32_G1_IBE_EN)
+	 (SIUL2_MSCR_S32_G1_IBE)
 
-#define SIUL2_MSCR_S32_G1_ENET_RX_ER_IN		SIUL2_MSCR_MUX_MODE_ALT2
+#define SIUL2_MSCR_S32_G1_ENET_RX_ER_IN		SIUL2_MSCR_S32_G1_MUX_ID_2
 
 #define SIUL2_MSCR_S32_G1_ENET_TX_D0	\
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	  SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_ENET_TX_D1	\
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	  SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_ENET_TX_D2	\
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	  SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_ENET_TX_D3	\
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	  SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_ENET_TX_EN	\
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	  SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_ENET_RMII_CLK_REF_IP	\
-	 (SIUL2_MSCR_S32_G1_IBE_EN)
+	 (SIUL2_MSCR_S32_G1_IBE)
 
 /* S32G Eth PFE Settings*/
 /* MAC0 MII */
@@ -651,63 +610,63 @@
 #define PFE_MAGIC 0x0800
 /* PFE default ops*/
 #define SIUL2_MSCR_S32_G1_PFE_OUT \
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
 	 PFE_MAGIC | \
-	 SIUL2_MSCR_S32_G1_OBE_EN)
+	  SIUL2_MSCR_S32_G1_OBE)
 
 #define SIUL2_MSCR_S32_G1_PFE_IN \
-	(SIUL2_MSCR_S32_G1_IBE_EN)
+	(SIUL2_MSCR_S32_G1_IBE)
 
 /* QSPI configuration  */
 #define SIUL2_MSCR_S32_G1_QSPI_BASE \
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_S32_G1_IBE_EN | \
-	 SIUL2_MSCR_S32_G1_PUE_EN | \
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	  SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_IBE | \
+	 SIUL2_MSCR_S32_G1_PUE | \
 	 SIUL2_MSCR_S32_G1_SMC_DIS)
 
 #define SIUL2_MSCR_S32_G1_QSPI_CLK_BASE	\
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN)
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	  SIUL2_MSCR_S32_G1_OBE)
 
 #define SIUL2_MSCR_S32_G1_QSPI_CS_BASE \
-	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
-	 SIUL2_MSCR_S32_G1_OBE_EN | \
-	 SIUL2_MSCR_S32_G1_PUE_EN | \
-	 SIUL2_MSCR_S32_G1_PUS_EN | \
+	(SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ | \
+	  SIUL2_MSCR_S32_G1_OBE | \
+	 SIUL2_MSCR_S32_G1_PUE | \
+	 SIUL2_MSCR_S32_G1_PUS | \
 	 SIUL2_MSCR_S32_G1_SMC_DIS)
 
-#define SIUL2_MSCR_S32_G1_QSPI_CK2_MUX		SIUL2_MSCR_MUX_MODE_ALT1
-#define SIUL2_MSCR_S32_G1_QSPI_A_SCK_MUX	SIUL2_MSCR_MUX_MODE_ALT1
-#define SIUL2_MSCR_S32_G1_QSPI_B_SCK_MUX	SIUL2_MSCR_MUX_MODE_ALT2
-#define SIUL2_MSCR_S32_G1_QSPI_A_CS0_MUX	SIUL2_MSCR_MUX_MODE_ALT1
-#define SIUL2_MSCR_S32_G1_QSPI_B_CS0_MUX	SIUL2_MSCR_MUX_MODE_ALT2
-#define SIUL2_MSCR_S32_G1_QSPI_A_CS1_MUX	SIUL2_MSCR_MUX_MODE_ALT1
-#define SIUL2_MSCR_S32_G1_QSPI_B_CS1_MUX	SIUL2_MSCR_MUX_MODE_ALT2
-#define SIUL2_IMCR_S32_G1_QSPI_A_DATA_MUX	SIUL2_MSCR_MUX_MODE_ALT2
-#define SIUL2_IMCR_S32_G1_QSPI_B_DATA_MUX	SIUL2_MSCR_MUX_MODE_ALT2
-#define SIUL2_MSCR_S32_G1_QSPI_A_DQS_MUX	SIUL2_MSCR_MUX_MODE_ALT2
+#define SIUL2_MSCR_S32_G1_QSPI_CK2_MUX		SIUL2_MSCR_S32_G1_MUX_ID_1
+#define SIUL2_MSCR_S32_G1_QSPI_A_SCK_MUX	SIUL2_MSCR_S32_G1_MUX_ID_1
+#define SIUL2_MSCR_S32_G1_QSPI_B_SCK_MUX	SIUL2_MSCR_S32_G1_MUX_ID_2
+#define SIUL2_MSCR_S32_G1_QSPI_A_CS0_MUX	SIUL2_MSCR_S32_G1_MUX_ID_1
+#define SIUL2_MSCR_S32_G1_QSPI_B_CS0_MUX	SIUL2_MSCR_S32_G1_MUX_ID_2
+#define SIUL2_MSCR_S32_G1_QSPI_A_CS1_MUX	SIUL2_MSCR_S32_G1_MUX_ID_1
+#define SIUL2_MSCR_S32_G1_QSPI_B_CS1_MUX	SIUL2_MSCR_S32_G1_MUX_ID_2
+#define SIUL2_IMCR_S32_G1_QSPI_A_DATA_MUX	SIUL2_MSCR_S32_G1_MUX_ID_2
+#define SIUL2_IMCR_S32_G1_QSPI_B_DATA_MUX	SIUL2_MSCR_S32_G1_MUX_ID_2
+#define SIUL2_MSCR_S32_G1_QSPI_A_DQS_MUX	SIUL2_MSCR_S32_G1_MUX_ID_2
 
 #define SIUL2_MSCR_S32_G1_QSPI_A_DQS \
 	(SIUL2_MSCR_S32_G1_QSPI_BASE | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_QSPI_A_DATA0_7 \
 	(SIUL2_MSCR_S32_G1_QSPI_BASE | \
-	 SIUL2_MSCR_S32_G1_PUS_EN | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	 SIUL2_MSCR_S32_G1_PUS | \
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_QSPI_A_CLK \
 	(SIUL2_MSCR_S32_G1_QSPI_CLK_BASE | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_QSPI_A_CS \
 	(SIUL2_MSCR_S32_G1_QSPI_CLK_BASE | \
-	 SIUL2_MSCR_MUX_MODE_ALT1)
+	 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_QSPI_B_DATA0_7 \
 	(SIUL2_MSCR_S32_G1_QSPI_BASE | \
-	 SIUL2_MSCR_MUX_MODE_ALT2)
+	 SIUL2_MSCR_S32_G1_MUX_ID_2)
 
 /* QSPI configuration */
 /* Note as-is DQS not enabled OR PAD_CTL_QSPI_A_DQS to enable */
@@ -777,12 +736,12 @@
 #define SIUL2_MSCR_S32_G1_PG2__QSPI_2A_B_SCK      99
 
 /* USB Configurations */
-#define SIUL2_MSCR_S32G_PAD_CTL_USB_DATA	(SIUL2_MSCR_S32_G1_OBE_EN | \
-						 SIUL2_MSCR_S32_G1_IBE_EN | \
-						 SIUL2_MSCR_MUX_MODE_ALT1)
-#define SIUL2_IMCR_S32G_PAD_CTL_USB_DATA	(SIUL2_MSCR_S32_G1_IBE_EN)
-#define SIUL2_MSCR_S32G_PAD_CTL_USB_STP		(SIUL2_MSCR_S32_G1_OBE_EN | \
-						 SIUL2_MSCR_MUX_MODE_ALT1)
+#define SIUL2_MSCR_S32G_PAD_CTL_USB_DATA	( SIUL2_MSCR_S32_G1_OBE | \
+						 SIUL2_MSCR_S32_G1_IBE | \
+						 SIUL2_MSCR_S32_G1_MUX_ID_1)
+#define SIUL2_IMCR_S32G_PAD_CTL_USB_DATA	(SIUL2_MSCR_S32_G1_IBE)
+#define SIUL2_MSCR_S32G_PAD_CTL_USB_STP		( SIUL2_MSCR_S32_G1_OBE | \
+						 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
 #define SIUL2_MSCR_S32_G1_PD14_USB_ULPI_DATA_O0		62
 #define SIUL2_MSCR_S32_G1_PD14_USB_ULPI_DATA_I0_IN	(896 - 512)
@@ -841,81 +800,81 @@
 #define SIUL2_MSCR_S32G_PF_15	95
 
 #define SIUL2_MSCR_S32_G1_PAD_CTL_SPI_CSx \
-	(SIUL2_MSCR_S32_G1_OBE_EN | SIUL2_MSCR_S32_G1_PUE_EN | \
-	 SIUL2_MSCR_S32_G1_PUS_EN)
+	( SIUL2_MSCR_S32_G1_OBE | SIUL2_MSCR_S32_G1_PUE | \
+	 SIUL2_MSCR_S32_G1_PUS)
 
 #define SUIL2_MSCR_S32G_PAD_CTL_SPI0_CS0 \
 	(SIUL2_MSCR_S32_G1_PAD_CTL_SPI_CSx | \
-	 SIUL2_MSCR_MUX_MODE_ALT2)
+	 SIUL2_MSCR_S32_G1_MUX_ID_2)
 
 #define SUIL2_MSCR_S32G_PAD_CTL_SPI0_CS1 \
 	(SIUL2_MSCR_S32_G1_PAD_CTL_SPI_CSx | \
-	 SIUL2_MSCR_MUX_MODE_ALT5)
+	 SIUL2_MSCR_S32_G1_MUX_ID_5)
 
 #define SUIL2_MSCR_S32G_PAD_CTL_SPI0_CS2 \
 	(SIUL2_MSCR_S32_G1_PAD_CTL_SPI_CSx | \
-	 SIUL2_MSCR_MUX_MODE_ALT3)
+	 SIUL2_MSCR_S32_G1_MUX_ID_3)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI0_SCK	(SIUL2_MSCR_S32_G1_OBE_EN | \
-						 SIUL2_MSCR_MUX_MODE_ALT1)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI0_SCK	( SIUL2_MSCR_S32_G1_OBE | \
+						 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI0_SOUT	(SIUL2_MSCR_S32_G1_OBE_EN | \
-						 SIUL2_MSCR_MUX_MODE_ALT1)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI0_SOUT	( SIUL2_MSCR_S32_G1_OBE | \
+						 SIUL2_MSCR_S32_G1_MUX_ID_1)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI0_SIN	(SIUL2_MSCR_S32_G1_IBE_EN | \
-						 SIUL2_MSCR_S32_G1_PUE_EN | \
-						 SIUL2_MSCR_S32_G1_PUS_EN)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI0_SIN	(SIUL2_MSCR_S32_G1_IBE | \
+						 SIUL2_MSCR_S32_G1_PUE | \
+						 SIUL2_MSCR_S32_G1_PUS)
 
 #define SUIL2_MSCR_S32G_PAD_CTL_SPI1_CS0 \
 	(SIUL2_MSCR_S32_G1_PAD_CTL_SPI_CSx | \
-	 SIUL2_MSCR_MUX_MODE_ALT2)
+	 SIUL2_MSCR_S32_G1_MUX_ID_2)
 
 #define SUIL2_MSCR_S32G_PAD_CTL_SPI1_CS3 \
 	(SIUL2_MSCR_S32_G1_PAD_CTL_SPI_CSx | \
-	 SIUL2_MSCR_MUX_MODE_ALT3)
+	 SIUL2_MSCR_S32_G1_MUX_ID_3)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI1_SCK	(SIUL2_MSCR_S32_G1_OBE_EN | \
-						 SIUL2_MSCR_MUX_MODE_ALT3)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI1_SCK	( SIUL2_MSCR_S32_G1_OBE | \
+						 SIUL2_MSCR_S32_G1_MUX_ID_3)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI1_SOUT	(SIUL2_MSCR_S32_G1_OBE_EN | \
-						 SIUL2_MSCR_MUX_MODE_ALT2)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI1_SOUT	( SIUL2_MSCR_S32_G1_OBE | \
+						 SIUL2_MSCR_S32_G1_MUX_ID_2)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI1_SIN	(SIUL2_MSCR_S32_G1_IBE_EN | \
-						 SIUL2_MSCR_S32_G1_PUE_EN | \
-						 SIUL2_MSCR_S32_G1_PUS_EN)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI1_SIN	(SIUL2_MSCR_S32_G1_IBE | \
+						 SIUL2_MSCR_S32_G1_PUE | \
+						 SIUL2_MSCR_S32_G1_PUS)
 
 #define SUIL2_MSCR_S32G_PAD_CTL_SPI4_CS0 \
 	(SIUL2_MSCR_S32_G1_PAD_CTL_SPI_CSx | \
-	 SIUL2_MSCR_MUX_MODE_ALT2)
+	 SIUL2_MSCR_S32_G1_MUX_ID_2)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI4_SCK	(SIUL2_MSCR_S32_G1_OBE_EN | \
-						 SIUL2_MSCR_MUX_MODE_ALT4)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI4_SCK	( SIUL2_MSCR_S32_G1_OBE | \
+						 SIUL2_MSCR_S32_G1_MUX_ID_4)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI4_SOUT	(SIUL2_MSCR_S32_G1_OBE_EN | \
-						 SIUL2_MSCR_MUX_MODE_ALT4)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI4_SOUT	( SIUL2_MSCR_S32_G1_OBE | \
+						 SIUL2_MSCR_S32_G1_MUX_ID_4)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI4_SIN	(SIUL2_MSCR_S32_G1_IBE_EN | \
-						 SIUL2_MSCR_S32_G1_PUE_EN | \
-						 SIUL2_MSCR_S32_G1_PUS_EN)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI4_SIN	(SIUL2_MSCR_S32_G1_IBE | \
+						 SIUL2_MSCR_S32_G1_PUE | \
+						 SIUL2_MSCR_S32_G1_PUS)
 
 #define SUIL2_MSCR_S32G_PAD_CTL_SPI5_CS0 \
 	(SIUL2_MSCR_S32_G1_PAD_CTL_SPI_CSx | \
-	 SIUL2_MSCR_MUX_MODE_ALT3)
+	 SIUL2_MSCR_S32_G1_MUX_ID_3)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI5_SCK	(SIUL2_MSCR_S32_G1_OBE_EN | \
-						 SIUL2_MSCR_MUX_MODE_ALT3)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI5_SCK	( SIUL2_MSCR_S32_G1_OBE | \
+						 SIUL2_MSCR_S32_G1_MUX_ID_3)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI5_SOUT	(SIUL2_MSCR_S32_G1_OBE_EN | \
-						 SIUL2_MSCR_MUX_MODE_ALT3)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI5_SOUT	( SIUL2_MSCR_S32_G1_OBE | \
+						 SIUL2_MSCR_S32_G1_MUX_ID_3)
 
-#define SIUL2_MSCR_S32G_PAD_CTL_SPI5_SIN	(SIUL2_MSCR_S32_G1_IBE_EN | \
-						 SIUL2_MSCR_S32_G1_PUE_EN | \
-						 SIUL2_MSCR_S32_G1_PUS_EN)
+#define SIUL2_MSCR_S32G_PAD_CTL_SPI5_SIN	(SIUL2_MSCR_S32_G1_IBE | \
+						 SIUL2_MSCR_S32_G1_PUE | \
+						 SIUL2_MSCR_S32_G1_PUS)
 
-#define SIUL2_IMCR_S32G_PAD_CTL_SPI0_SIN	(SIUL2_MSCR_MUX_MODE_ALT2)
-#define SIUL2_IMCR_S32G_PAD_CTL_SPI1_SIN	(SIUL2_MSCR_MUX_MODE_ALT2)
-#define SIUL2_IMCR_S32G_PAD_CTL_SPI4_SIN	(SIUL2_MSCR_MUX_MODE_ALT2)
-#define SIUL2_IMCR_S32G_PAD_CTL_SPI5_SIN	(SIUL2_MSCR_MUX_MODE_ALT2)
+#define SIUL2_IMCR_S32G_PAD_CTL_SPI0_SIN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define SIUL2_IMCR_S32G_PAD_CTL_SPI1_SIN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define SIUL2_IMCR_S32G_PAD_CTL_SPI4_SIN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define SIUL2_IMCR_S32G_PAD_CTL_SPI5_SIN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
 
 #define SIUL2_PA_14_IMCR_S32G_SPI0_SIN		(982 - 512)
 #define SIUL2_PF_15_IMCR_S32G_SPI1_SIN		(987 - 512)
@@ -965,13 +924,13 @@
 #define SIUL2_GPIO_VALUE1	(0x01)
 
 #define SIUL2_MSCR_GPI_G1 \
-	(SIUL2_MSCR_MUX_MODE_ALT0 | \
-	 SIUL2_MSCR_S32_G1_IBE_EN | \
-	 SIUL2_MSCR_S32_G1_PUS_EN | \
-	 SIUL2_MSCR_S32_G1_PUE_EN)
+	(SIUL2_MSCR_S32_G1_MUX_ID_0 | \
+	 SIUL2_MSCR_S32_G1_IBE | \
+	 SIUL2_MSCR_S32_G1_PUS | \
+	 SIUL2_MSCR_S32_G1_PUE)
 
 #define SIUL2_MSCR_GPO_G1 \
-	(SIUL2_MSCR_MUX_MODE_ALT0 | \
-	 SIUL2_MSCR_S32_G1_OBE_EN)
+	(SIUL2_MSCR_S32_G1_MUX_ID_0 | \
+	  SIUL2_MSCR_S32_G1_OBE)
 
 #endif /*__ARCH_ARM_MACH_S32G1_SIUL_H__ */
diff --git a/drivers/gpio/gpio-s32.c b/drivers/gpio/gpio-s32.c
index 33f878a7c5..afc050ccea 100644
--- a/drivers/gpio/gpio-s32.c
+++ b/drivers/gpio/gpio-s32.c
@@ -11,16 +11,12 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <errno.h>
 
+#include <dt-bindings/pinctrl/s32-gen1-pinctrl.h>
+
 #define MSCR_OFF	0x0240
 #define GPDO_BASE	0x1300
 #define GPDI_BASE	0x1500
 
-/* S32-GEN1 SIUL2_MSCR masks */
-#define SIUL2_MSCR_S32_G1_OBE		BIT(21)
-#define SIUL2_MSCR_S32_G1_ODE		BIT(20)
-#define SIUL2_MSCR_S32_G1_IBE		BIT(19)
-#define SIUL2_MSCR_S32_G1_SSS_MASK	GENMASK(2, 0)
-
 #define SIUL2_MAX_VALID_RANGES		4
 
 struct gpio_range {
diff --git a/include/dt-bindings/pinctrl/s32-gen1-pinctrl.h b/include/dt-bindings/pinctrl/s32-gen1-pinctrl.h
index 4fb31bbeb4..36ff2090bd 100644
--- a/include/dt-bindings/pinctrl/s32-gen1-pinctrl.h
+++ b/include/dt-bindings/pinctrl/s32-gen1-pinctrl.h
@@ -6,16 +6,15 @@
 #ifndef S32_GEN1_PINCTRL_H
 #define S32_GEN1_PINCTRL_H
 
-#ifndef BIT
-#define BIT(nr)		(1U << (nr))
-#endif
-
-#define SIUL2_MSCR_S32_G1_OBE		BIT(21) /* Output buffer enable. */
-#define SIUL2_MSCR_S32_G1_ODE		BIT(20) /* Open drain eable. */
-#define SIUL2_MSCR_S32_G1_IBE		BIT(19) /* Input buffer enable. */
-#define SIUL2_MSCR_S32_G1_PUE		BIT(13) /* Pull enable. */
-#define SIUL2_MSCR_S32_G1_PUS		BIT(12) /* Pull-up enalbe. */
-#define SIUL2_MSCR_S32_G1_SMC_DIS	BIT(5) /* Safe mode conrol disable*/
+#define SIUL2_MSCR_S32_G1_OBE		(1 << 21) /* Output buffer enable. */
+#define SIUL2_MSCR_S32_G1_ODE		(1 << 20) /* Open drain eable. */
+#define SIUL2_MSCR_S32_G1_IBE		(1 << 19) /* Input buffer enable. */
+#define SIUL2_MSCR_S32_G1_PUE		(1 << 13) /* Pull enable. */
+#define SIUL2_MSCR_S32_G1_PUS		(1 << 12) /* Pull-up enalbe. */
+#define SIUL2_MSCR_S32_G1_SMC_DIS	(1 << 5) /* Safe mode conrol disable. */
+#define SIUL2_MSCR_S32_G1_SSS_MASK	(7 << 0) /* Source sig. select. */
+#define SIUL2_MSCR_S32_G1_RCVR_DBL	(0 << 10)
+#define SIUL2_MSCR_S32_G1_RCVR_SNGL	(1 << 10)
 
 #define SIUL2_MSCR_S32_G1_SRC_208_1V8_166_3V3_MHZ	(0 << 14)
 #define SIUL2_MSCR_S32_G1_SRC_166_1V8_150_3V3_MHZ	(4 << 14)
-- 
2.17.1

